fineract-provider/src/main/java/org/apache/fineract/infrastructure/creditbureau/api/CreditBureauIntegrationAPI.java

```diff
@@ -0,0 +1,160 @@
+            final DefaultToApiJsonSerializer<CreditReportData> toCreditReportApiJsonSerializer,
+            final PortfolioCommandSourceWritePlatformService commandsSourceWritePlatformService,
+            final ApiRequestParameterHelper apiRequestParameterHelper,
+            final CreditReportWritePlatformService creditReportWritePlatformService,
+            final CreditReportReadPlatformService creditReportReadPlatformService,
+            final DefaultToApiJsonSerializer<CreditReportData> toApiJsonSerializer) {
+        this.context = context;
+        this.toCreditReportApiJsonSerializer = toCreditReportApiJsonSerializer;
+        this.commandsSourceWritePlatformService = commandsSourceWritePlatformService;
+        this.apiRequestParameterHelper = apiRequestParameterHelper;
+        this.creditReportWritePlatformService = creditReportWritePlatformService;
+        this.creditReportReadPlatformService = creditReportReadPlatformService;
+        this.toApiJsonSerializer = toApiJsonSerializer;
+
+    }
+
+    @POST
+    @Path("creditReport")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    public String postCreditReport(@Context final UriInfo uriInfo, @RequestParam("params") final Map<String, Object> params) {
+
+        Gson gson = new Gson();
+        final String json = gson.toJson(params);
+        final CommandWrapper commandRequest = new CommandWrapperBuilder().getCreditReport().withJson(json).build();
+
+        final CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
+        return this.toCreditReportApiJsonSerializer.serialize(result);
+
+    }
+
+    // saves fetched-creditreport into database
+    @POST
+    @Path("saveCreditReport")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    public String saveCreditReport(@Parameter(hidden = true) final String apiRequestBodyAsJson,
+            @QueryParam("creditBureauId") @Parameter(description = "creditBureauId") final Long creditBureauId,
+            @QueryParam("creditReportNumber") @Parameter(description = "creditReportNumber") final String creditReportNumber) {
+
+        final CommandWrapper commandRequest = new CommandWrapperBuilder() //
+                .saveCreditReport(creditBureauId, creditReportNumber) // creditReportNumber is a NRC number for
+                                                                      // Thitsawork
+                .withJson(apiRequestBodyAsJson) // apiRequestBodyAsJson is a creditReport
+                .build(); //
+
+        final CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
+        return this.toCreditReportApiJsonSerializer.serialize(result);
+
+    }
+
+    // fetch saved creditReports(NRC) from DB by creditBureauId, to select for downloading and deleting the reports
+    @GET
+    @Path("creditReport/{creditBureauId}")
```

Comment:
rename creditBureauID to creditReportID

Reply:
To get the details of the credit report specifically of that credit bureau, we are using creditBureauId, It fetches the credit report details like (national Id/NRC) to check whether that specific credit report has been stored or not in the database.